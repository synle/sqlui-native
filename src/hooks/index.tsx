import { useQuery } from 'react-query';

type ConnectionProps = {
  id: string;
  connection: string;
  name: string;
  [index: string]: any;
};

type SqlColumn = {
  type: string;
  allowNull: boolean;
  defaultValue?: string;
  comment?: string;
  special?: string;
  primaryKey: boolean;
};

// @ts-ignore
function _fetch<T>(...inputs) {
  return fetch(inputs[0], {
    ...(inputs[1] || {}),
    'content-type': 'Application/JSON',
  })
    .then((r) => r.json())
    .then((r) => {
      const res: T = r;
      return res;
    });
}

// useQuery('repoData', () =>
//    fetch('https://api.github.com/repos/tannerlinsley/react-query').then(res =>
//      res.json()
//    )
//  )

export function useGetConnections() {
  return useQuery(['connection'], () => _fetch<ConnectionProps[]>(`/api/connections`));
}

export function useGetConnection(connectionId: string) {
  return useQuery(['connection', connectionId], () => _fetch(`/api/connection/${connectionId}`));
}

export function useAddConnection() {}

export function useUpdateConnection() {}

export function useDeleteConnection() {}

export function useGetDatabases(connectionId: string) {
  return useQuery(['connection', connectionId, 'databases'], () =>
    _fetch<string[]>(`/api/connection/${connectionId}/databases`),
  );
}

export function useGetTables(connectionId: string, databaseId: string) {
  return useQuery(['connection', connectionId, 'database', databaseId, 'tables'], () =>
    _fetch<string[]>(`/api/connection/${connectionId}/database/${databaseId}/tables`),
  );
}

export function useGetColumns(connectionId: string, databaseId: string, tableId: string) {
  return useQuery(['connection', connectionId, 'database', databaseId, 'table', tableId], () =>
    _fetch<{ [index: string]: SqlColumn }>(
      `/api/connection/${connectionId}/database/${databaseId}/table/${tableId}/columns`,
    ),
  );
}

export function useExecute(connectionId: string, sql: string, databaseId?: string) {
  return useQuery(['connection', connectionId, 'database', databaseId, 'table'], () =>
    _fetch(`/api/connection/${connectionId}/execute`, {
      data: {
        database: databaseId,
        sql,
      },
    }),
  );
}
