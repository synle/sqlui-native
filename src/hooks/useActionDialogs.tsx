import {useQuery} from 'react-query';
import {useQueryClient} from 'react-query';
import React from 'react';
import {ChoiceInput} from 'src/components/ActionDialogs/ChoiceDialog';
import {ChoiceOption} from 'src/components/ActionDialogs/ChoiceDialog';
import {ModalInput} from 'src/components/ActionDialogs/ModalDialog';
import {PromptInput} from 'src/components/ActionDialogs/PromptDialog';
type AlertActionDialog = {
  type: 'alert';
  message: string;
  onSubmit?: () => void;
};

type ConfirmActionDialog = {
  type: 'confirm';
  message: string;
  onSubmit: (yesSelected: boolean) => void;
};

type ChoiceActionDialog = ChoiceInput & {
  type: 'choice';
  onSubmit: (yesSelected: boolean, selectedChoice?: string) => void;
};

type PromptActionDialog = PromptInput & {
  type: 'prompt';
  onSubmit: (yesSelected: boolean, newValue?: string) => void;
};

type ModalActionDialog = ModalInput & {
  type: 'modal';
  onSubmit: (closed: boolean) => void;
};

type ActionDialog =
  | AlertActionDialog
  | ConfirmActionDialog
  | PromptActionDialog
  | ChoiceActionDialog
  | ModalActionDialog;

const QUERY_KEY_ACTION_DIALOGS = 'actionDialogs';
let _actionDialogs: ActionDialog[] = [];

export function useActionDialogs() {
  const queryClient = useQueryClient();

  const { data, isLoading: loading } = useQuery(QUERY_KEY_ACTION_DIALOGS, () => _actionDialogs);

  const prompt = (props: PromptInput): Promise<string | undefined> => {
    return new Promise((resolve, reject) => {
      const { message, value, isLongPrompt } = props;

      const newActionDialog: ActionDialog = {
        ...props,
        type: 'prompt',
        onSubmit: (yesSelected, newValue) => {
          yesSelected ? resolve(newValue) : reject();
        },
      };
      _actionDialogs.push(newActionDialog);
      queryClient.invalidateQueries(QUERY_KEY_ACTION_DIALOGS);
    });
  };

  const confirm = (message: string): Promise<void> => {
    return new Promise((resolve, reject) => {
      const newActionDialog: ActionDialog = {
        type: 'confirm',
        message,
        onSubmit: (yesSelected) => {
          yesSelected ? resolve() : reject();
        },
      };
      _actionDialogs.push(newActionDialog);
      queryClient.invalidateQueries(QUERY_KEY_ACTION_DIALOGS);
    });
  };

  const choice = (
    title: string,
    message: string | React.ReactNode,
    options: ChoiceOption[],
  ): Promise<string> => {
    return new Promise((resolve, reject) => {
      const newActionDialog: ActionDialog = {
        type: 'choice',
        title,
        message,
        options,
        onSubmit: (yesSelected, newValue) => {
          yesSelected && newValue ? resolve(newValue) : reject();
        },
      };
      _actionDialogs.push(newActionDialog);
      queryClient.invalidateQueries(QUERY_KEY_ACTION_DIALOGS);
    });
  };

  const alert = (message: string): Promise<void> => {
    return new Promise((resolve, reject) => {
      const newActionDialog: ActionDialog = {
        type: 'alert',
        message,
      };
      _actionDialogs.push(newActionDialog);
      queryClient.invalidateQueries(QUERY_KEY_ACTION_DIALOGS);
    });
  };

  const modal = (props: ModalInput): Promise<void> => {
    return new Promise((resolve, reject) => {
      _actionDialogs.push({
        type: 'modal',
        onSubmit: () => {},
        ...props,
      });
      queryClient.invalidateQueries(QUERY_KEY_ACTION_DIALOGS);
    });
  };

  let dialog;
  try {
    if (data) {
      dialog = data[data.length - 1];
    }
  } catch (err) {
    dialog = undefined;
  }

  const dismiss = () => {
    _actionDialogs.pop();
    queryClient.invalidateQueries(QUERY_KEY_ACTION_DIALOGS);
  };

  return {
    dialogs: data,
    dialog,
    alert,
    prompt,
    confirm,
    choice,
    dismiss,
    modal,
  };
}