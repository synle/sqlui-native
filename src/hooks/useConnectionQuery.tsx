import { QueryClient, useQuery, useQueryClient } from 'react-query';
import dataApi from 'src/data/api';
import { SessionStorageConfig } from 'src/data/config';
import { getGeneratedRandomId, getUpdatedOrdersForList } from 'src/utils/commonUtils';
import { SqluiFrontend, SqluiCore } from 'typings';

const QUERY_KEY_QUERIES = 'queries';

// connection queries
let _connectionQueries: SqluiFrontend.ConnectionQuery[];

function _useConnectionQueries() {
  return useQuery(
    QUERY_KEY_QUERIES,
    async () => {
      if (!_connectionQueries) {
        // this is the first time
        // try pulling it in from sessionStorage
        _connectionQueries = SessionStorageConfig.get<SqluiFrontend.ConnectionQuery[]>(
          'clientConfig/cache.connectionQueries',
          [],
        );

        if (_connectionQueries.length === 0) {
          // if config failed, attempt to get it from the api
          try {
            _connectionQueries = await dataApi.getQueries();
          } catch (err) {}
        }

        // at the end we want to remove executionStart so the query won't be run again
        let toBeSelectedQuery = 0;
        _connectionQueries = _connectionQueries.map((query, idx) => {
          if (query.selected) {
            query.selected = false;
            toBeSelectedQuery = idx;
          }

          return query;
        });

        if (_connectionQueries[toBeSelectedQuery]) {
          _connectionQueries[toBeSelectedQuery].selected = true;
        }
      }

      return _connectionQueries;
    },
    {
      onSuccess: (data) => {
        SessionStorageConfig.set('clientConfig/cache.connectionQueries', data);
      },
    },
  );
}

export function useConnectionQueries() {
  const queryClient = useQueryClient();

  const { data: queries, isLoading, isFetching } = _useConnectionQueries();

  function _invalidateQueries() {
    queryClient.setQueryData<SqluiFrontend.ConnectionQuery[] | undefined>(
      QUERY_KEY_QUERIES,
      () => _connectionQueries,
    );
  }

  const onAddQuery = async (query?: SqluiCore.CoreConnectionQuery) => {
    const newId = getGeneratedRandomId(`queryId`);

    let newQuery: SqluiFrontend.ConnectionQuery;
    if (!query) {
      newQuery = {
        id: newId,
        name: `Query ${new Date().toLocaleString()}`,
        sql: '',
        selected: true,
      };
    } else {
      let newQueryName = query.name || `Query ${new Date().toLocaleString()}`;

      for (const query of _connectionQueries) {
        if (query.name === newQueryName) {
          // replace it with a new anme
          newQueryName = `Duplicated Query ${new Date().toLocaleString()}`;
        }
      }

      newQuery = {
        ...query,
        id: newId,
        name: newQueryName,
        selected: true,
      };
    }

    _connectionQueries = [
      ..._connectionQueries.map((q) => {
        q.selected = false;
        return q;
      }),
      newQuery,
    ];

    try {
      _invalidateQueries();
      dataApi.upsertQuery(newQuery);// make an api call to persists and this is fire and forget
    } catch (err) {}

    return newQuery;
  };

  const onDeleteQueries = (queryIds?: string[]) => {
    if (!queryIds || queryIds.length === 0) {
      return;
    }

    let toBeSelected = 0;
    if (queryIds.length === 1) {
      const [queryId] = queryIds;
      _connectionQueries = _connectionQueries.filter((q, idx) => {
        if (q.id !== queryId) {
          return true;
        }
        toBeSelected = Math.max(0, idx - 1);

        return false;
      });
    } else {
      _connectionQueries = _connectionQueries.filter((q, idx) => {
        if (queryIds.indexOf(q.id) >= 0) {
          return false;
        }

        return true;
      });
    }

    if (_connectionQueries[toBeSelected]) {
      _connectionQueries[toBeSelected].selected = true;
    }

    // make an api call to persists
    // this is fire and forget
    for (const queryId of queryIds) {
      // make api to delete the query
      dataApi.deleteQuery(queryId);
    }

    _invalidateQueries();
  };

  const onDeleteQuery = (queryId?: string) => queryId && onDeleteQueries([queryId]);

  const onShowQuery = (queryId: string) => {
    _connectionQueries = _connectionQueries.map((q) => {
      q.selected = q.id === queryId;
      return q;
    });
    _invalidateQueries();
  };

  const onChangeQuery = async (
    queryId: string | undefined,
    partials: SqluiFrontend.PartialConnectionQuery,
  ) => {
    if(!queryId){
      if(!queries || queries.length === 0){
        // this is an edge case where users already closed all the query tab
        const newQuery = await onAddQuery({
          id: getGeneratedRandomId(`queryId`),
          name: `Query ${new Date().toLocaleString()}`,
          ...partials
        });

        queryId = newQuery.id;
      }
    }

    const query = queries?.find((q) => q.id === queryId);

    if (!query || !query) {
      return;
    }

    _connectionQueries = [..._connectionQueries].map((query) => {
      if (query.id === queryId) {
        const newValue = {
          ...query,
          ...partials,
        };
        return newValue;
      }

      return query;
    });

    try {
      _invalidateQueries();
      dataApi.upsertQuery(query);// make an api call to persists and this is fire and forget
    } catch (err) {}
  };

  const onDuplicateQuery = (queryId?: string) => {
    const query = queries?.find((q) => q.id === queryId);

    if (!query || !query) {
      return;
    }

    onAddQuery(query);
  };

  const onImportQuery = (query?: SqluiFrontend.ConnectionQuery) => {
    if (!query || !query) {
      return;
    }

    return onAddQuery(query);
  };

  const onOrderingChange = (from: number, to: number) => {
    _connectionQueries = getUpdatedOrdersForList(_connectionQueries, from, to);
    _invalidateQueries();
  };

  return {
    isLoading,
    isFetching,
    queries,
    onAddQuery,
    onDeleteQuery,
    onDeleteQueries,
    onShowQuery,
    onChangeQuery,
    onDuplicateQuery,
    onImportQuery,
    onOrderingChange,
  };
}

export function useConnectionQuery(queryId: string) {
  const queryClient = useQueryClient();

  const { queries, onChangeQuery, onDeleteQuery, isLoading, isFetching } = useConnectionQueries();

  const query = queries?.find((q) => q.id === queryId);

  const onChange = (partials: SqluiFrontend.PartialConnectionQuery) =>
    onChangeQuery(query?.id, partials);

  const onDelete = () => onDeleteQuery(query?.id);

  return {
    isLoading,
    isFetching,
    query,
    onChange,
    onDelete,
  };
}

export function useActiveConnectionQuery() {
  const queryClient = useQueryClient();

  const { queries, onChangeQuery, onDeleteQuery, isLoading, isFetching } = useConnectionQueries();

  const query = queries?.find((q) => q.selected);

  const onChange = (partials: SqluiFrontend.PartialConnectionQuery) =>
    onChangeQuery(query?.id, partials);

  const onDelete = () => onDeleteQuery(query?.id);

  return {
    isLoading,
    isFetching,
    query,
    onChange,
    onDelete,
  };
}
