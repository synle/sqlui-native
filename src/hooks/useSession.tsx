import useMediaQuery from '@mui/material/useMediaQuery';
import { QueryClient } from 'react-query';
import { useMutation } from 'react-query';
import { useQuery } from 'react-query';
import { useQueryClient } from 'react-query';
import dataApi from 'src/data/api';
import { LocalStorageConfig } from 'src/data/config';
import { SessionStorageConfig } from 'src/data/config';
import { getCurrentSessionId } from 'src/data/session';
import { SqluiCore } from 'typings';
import { SqluiFrontend } from 'typings';

const QUERY_KEY_SESSIONS = 'qk.sessions';

// for sessions
export function useGetSessions() {
  return useQuery([QUERY_KEY_SESSIONS], dataApi.getSessions);
}

export function useGetCurrentSession() {
  const { data, ...rest } = useGetSessions();

  const currentMatchedSession = data?.find((session) => session.id === getCurrentSessionId());

  return {
    data: currentMatchedSession,
    ...rest,
  };
}

export function useUpsertSession() {
  const queryClient = useQueryClient();
  return useMutation<SqluiCore.Session, void, SqluiCore.CoreSession>(dataApi.upsertSession, {
    onSuccess: async (newSession) => {
      queryClient.invalidateQueries(QUERY_KEY_SESSIONS);
      return newSession;
    },
  });
}

export function useDeleteSession() {
  const queryClient = useQueryClient();

  return useMutation<string, void, string>(dataApi.deleteSession, {
    onSuccess: async (deletedSessionId) => {
      queryClient.invalidateQueries(QUERY_KEY_SESSIONS);
      return deletedSessionId;
    },
  });
}
