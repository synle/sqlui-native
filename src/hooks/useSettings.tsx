import useMediaQuery from '@mui/material/useMediaQuery';
import { QueryClient } from 'react-query';
import { useMutation } from 'react-query';
import { useQuery } from 'react-query';
import { useQueryClient } from 'react-query';
import dataApi from 'src/data/api';
import { LocalStorageConfig } from 'src/data/config';
import { SessionStorageConfig } from 'src/data/config';
import { getCurrentSessionId } from 'src/data/session';
import { SqluiCore } from 'typings';
import { SqluiFrontend } from 'typings';

const QUERY_KEY_SETTINGS = 'qk.settings';

// Settings
let _settings = LocalStorageConfig.get<SqluiFrontend.Settings>('clientConfig/cache.settings', {});

export function useSettings() {
  const queryClient = useQueryClient();

  const { data: settings, isLoading } = useQuery(QUERY_KEY_SETTINGS, () => _settings, {
    onSuccess: (data) => LocalStorageConfig.set('clientConfig/cache.settings', _settings),
  });

  const onChange = (newSettings: SqluiFrontend.Settings) => {
    _settings = { ...newSettings };

    queryClient.setQueryData<SqluiFrontend.Settings | undefined>(
      QUERY_KEY_SETTINGS,
      () => _settings,
    );
  };

  return {
    isLoading,
    settings,
    onChange,
  };
}

export function useDarkModeSetting() {
  const { settings } = useSettings();
  const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');

  let value = settings?.darkMode;

  if (!value) {
    value = prefersDarkMode ? 'dark' : 'light';
  }

  return value;
}

export function useEditorModeSetting() {
  const { settings } = useSettings();

  let value = settings?.editorMode;

  if (value !== 'simple') {
    value = 'advanced';
  }

  return value;
}

export function useWordWrapSetting() {
  const { settings } = useSettings();
  return settings?.wordWrap === 'wrap';
}

export function useQueryTabOrientationSetting() {
  const { settings } = useSettings();
  return settings?.queryTabOrientation;
}

export const DEFAULT_QUERY_SIZE = 100;

export function useQuerySizeSetting() {
  const { settings } = useSettings();
  return parseInt(settings?.querySize + '') || DEFAULT_QUERY_SIZE;
}
