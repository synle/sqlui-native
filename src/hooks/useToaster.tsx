import { useQuery } from 'react-query';
import { useQueryClient } from 'react-query';
import { AnchorOrigin } from 'src/components/Toast';
import Toast from 'src/components/Toast';

const QUERY_KEY_TOASTS = 'toasts';

type CoreToasterProps = {
  onClose?: () => void;
  message: string;
  anchorOrigin?: AnchorOrigin;
  autoHideDuration?: number;
};

type ToasterProps = CoreToasterProps & {
  id: string;
};

type ToasterHandler = {
  dismiss: (dismissDelay?: number) => void;
};

let _toasts: ToasterProps[] = [];

const DEFAULT_AUTO_HIDE_DURATION = 6000;

export default function useToaster() {
  const queryClient = useQueryClient();

  const { data, isLoading } = useQuery(QUERY_KEY_TOASTS, () => _toasts);

  const add = (props: CoreToasterProps): Promise<ToasterHandler> => {
    return new Promise((resolve, reject) => {
      const toastId = `toast.${Date.now()}.${Math.floor(Math.random() * 10000000000000000)}`;
      _toasts.push({
        ...props,
        id: toastId,
        autoHideDuration: props.autoHideDuration || DEFAULT_AUTO_HIDE_DURATION,
      });
      queryClient.invalidateQueries(QUERY_KEY_TOASTS);

      resolve({
        dismiss: (dismissDelay?: number) => {
          dismiss(toastId, dismissDelay)
        },
      });
    });
  };

  const dismiss = (toastId: string, dismissDelay?: number): Promise<void> => {
    return new Promise((resolve, reject) => {
      setTimeout(() => {
        _toasts = _toasts.filter((toast) => toast.id !== toastId);
        queryClient.invalidateQueries(QUERY_KEY_TOASTS);
        resolve();
      }, dismissDelay || 0);
    });
  };

  let toast;
  try {
    if (data) {
      toast = data[data.length - 1];
    }
  } catch (err) {
    toast = undefined;
  }

  return {
    toasts: data,
    toast,
    isLoading,
    add,
    dismiss,
  };
}
