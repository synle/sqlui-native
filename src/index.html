<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>SQL UI Native</title>
  <meta name="viewport" content="width=device-width, initial-scale=1">
</head>

<body>
  <div id="main"></div>
  <script type="module" src="./index.tsx"></script>
</body>

</html>
