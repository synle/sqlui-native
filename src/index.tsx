import CssBaseline from '@mui/material/CssBaseline';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import { SnackbarProvider } from 'notistack';
import ReactDOM from 'react-dom';
import { QueryClient, QueryClientProvider } from 'react-query';
import { ReactQueryDevtools } from 'react-query/devtools';
import { HashRouter, Route, Routes } from 'react-router-dom';
import React from 'react';
import App from 'src/frontend/App';
import ActionDialogs from 'src/frontend/components/ActionDialogs';
import DataItemView from 'src/frontend/DataItemView';
import ActionDialogsContextProvider from 'src/frontend/hooks/useActionDialogs';
import ConnectionQueryContextProvider from 'src/frontend/hooks/useConnectionQuery';
import SettingContextProvider, { useDarkModeSetting } from 'src/frontend/hooks/useSetting';
import ShowHideContextProvider from 'src/frontend/hooks/useShowHide';
import TreeActionContextProvider from 'src/frontend/hooks/useTreeActions';
import 'src/frontend/App.scss';
import 'src/frontend/electronRenderer';

function AppliedTheme({ children }) {
  const myTheme = createTheme({
    palette: {
      mode: useDarkModeSetting(),
    },
    components: {
      MuiButtonBase: {
        defaultProps: {
          disableRipple: true,
        },
      },
    },
  });

  return <ThemeProvider theme={myTheme}>{children}</ThemeProvider>;
}

function CombinedContextProvider({ children }) {
  return [
    ActionDialogsContextProvider,
    ConnectionQueryContextProvider,
    TreeActionContextProvider,
    ShowHideContextProvider,
    SettingContextProvider,
    HashRouter,
    AppliedTheme,
  ].reduceRight((acc, Provider) => <Provider>{acc}</Provider>, children);
}

const renderApp = function () {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
        refetchOnWindowFocus: false,
      },
    },
  });

  ReactDOM.render(
    <SnackbarProvider maxSnack={4}>
      <QueryClientProvider client={queryClient}>
        <CssBaseline />
        <ReactQueryDevtools initialIsOpen={false} />

        <CombinedContextProvider>
          <Routes>
            <Route path='/data-table/:dataItemGroupKey' element={<DataItemView />} />
            <Route path='/*' element={<App />} />
          </Routes>
        </CombinedContextProvider>
      </QueryClientProvider>
    </SnackbarProvider>,
    document.querySelector('#body'),
  );

  window.removeEventListener('sqluiNativeEvent/ready', renderApp);
};

window.addEventListener('sqluiNativeEvent/ready', renderApp, false);

// tell the main app to get ready for initiation
window.dispatchEvent(new Event('sqluiNativeEvent/init'));
