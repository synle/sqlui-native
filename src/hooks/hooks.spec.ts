import * as hooks from 'src/hooks';

describe('hooks', () => {
  describe('getExportedConnection', () => {
    test('should work', async () => {
      const actual = hooks.getExportedConnection({
        id: 'connection.1643485467072.6333713976068809',
        connection: 'mysql://root:password@localhost:3306',
        name: 'sy mysql 123',
        status: 'online',
        dialect: 'mysql',
      });
      expect(actual).toMatchSnapshot();
    });
  });

  describe('getExportedQuery', () => {
    test('should work', async () => {
      const actual = hooks.getExportedQuery({
        id: 'query.1643737746323.6184509846791006',
        name: 'Query 2/1/2022, 9:49:06 AM',
        sql: 'SELECT\n  TOP 200 *\nFROM\n  albums',
        connectionId: 'connection.1644098335891.8887562323718656',
        databaseId: 'musicstores',
      });
      expect(actual).toMatchSnapshot();
    });
  });
});
