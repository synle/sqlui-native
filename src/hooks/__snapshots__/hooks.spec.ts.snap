// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hooks getExportedConnection should work 1`] = `
Object {
  "_type": "connection",
  "connection": "mysql://root:password@localhost:3306",
  "id": "connection.1643485467072.6333713976068809",
  "name": "sy mysql 123",
}
`;

exports[`hooks getExportedQuery should work 1`] = `
Object {
  "_type": "query",
  "connectionId": "connection.1644098335891.8887562323718656",
  "databaseId": "musicstores",
  "id": "query.1643737746323.6184509846791006",
  "name": "Query 2/1/2022, 9:49:06 AM",
  "sql": "SELECT
  TOP 200 *
FROM
  albums",
}
`;
